larisa.py

```diff
@@ -8,59 +8,82 @@
 import requests
 from geopy import Nominatim
 
-def weather_forecast(city):
+
+def temperature_conversion(temp):
+    '''
+    #Author: Nils Becker
+    Converts a given temperature in °K into a °C String representation
+    :param temp: Float Value of temperature in °K
+    :returns: The temperature converted to °C as a String
+    '''
+    return f"{int(temp)-273}°C"
+
+def weather_forecast(cfg, city):
     '''
     #Author: Nils Becker
     Gives a weather forecast for a specified city
+    :param cfg: Currently used configuration
     :param city: String representation of city
     :returns: weather forecast for now/ next 12h/ next 7 days in a List
     '''
     locator = Nominatim(user_agent="Larissa")
     location = locator.geocode(city)
-    api_key = '14ebab67cd0768d48dd1eb7545d89ac1'
-    url = 'https://api.openweathermap.org/data/2.5/onecall?lat={}&lon={}&exclude={}&appid={}'.format(location.latitude, location.longitude, "minutely,alerts", api_key)
-    response = requests.get(url).json()
+    api_key = cfg['weather']['apikey']
+    try:
+    #create API request for given city
+        url = 'https://api.openweathermap.org/data/2.5/onecall?lat={}&lon={}&exclude={}&appid={}'.format(location.latitude, location.longitude, "minutely,alerts", api_key)
+        response = requests.get(url).json()
+    except AttributeError:
+        logging.info("City not found")
+        return None
     result = []
+    try:
+    #get data
+        current = response['current']
+        currentList = []
+        temperature = temperature_conversion(current['temp'])
+        weather = current['weather']
+        description = weather[0]['description']
+        clouds = str(current['clouds']) + "%"+" clouds"

```

Comment:
spaces. Or even better `clouds = f"{current['clouds']}% clouds"`
And everywhere you build a string